### PR TITLE
FAT-816: Implement e-2-e automation of test case C9215

### PR DESCRIPTION
**Pull request by mariia-aloshyna**

### CODE REVIEW

File: cypress/integration/inventory/assign-preceding-title.spec.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+          {
+            instanceTypeId: Cypress.env('instanceTypes')[0].id,
+            title: instanceTitle,
+            source: 'FOLIO',
+          }, {
+            instanceTypeId: Cypress.env('instanceTypes')[0].id,
+            title: instanceTitle2,
+            source: 'FOLIO',
+          }
+        ]).each(instance => cy.createInstance({ instance }));
```
Comment (MaksimTerekhov) [inline cypress/integration/inventory/assign-preceding-title.spec.js:33]: please add "after" with deleting your test data

File: cypress/integration/inventory/assign-preceding-title.spec.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import getRandomPostfix from '../../support/utils/stringTools';
+import InventorySearch from '../../support/fragments/inventory/inventorySearch';
+import InventoryInstances from '../../support/fragments/inventory/inventoryInstances';
+import InventoryInstance from '../../support/fragments/inventory/inventoryInstance';
+import InventoryInstanceEdit from '../../support/fragments/inventory/InventoryInstanceEdit';
+
+describe('ui-inventory: Assign a Preceding title for an instance', () => {
+  const instanceTitle = `autoTestInstanceTitle.${getRandomPostfix()}`;
+  const instanceTitle2 = `autoTestInstanceTitle.${getRandomPostfix()}`;
+  const precedingTitleValue = 'Preceding title test value';
```
Comment (MaksimTerekhov) [inline cypress/integration/inventory/assign-preceding-title.spec.js:11]: Better to use getRandomPostfix() here
It helps to debug esilier if test will be failed in future

File: cypress/support/fragments/inventory/InventoryInstanceEdit.js [hunk 3]
```diff
@@ -82,7 +85,27 @@ export default {
...
+  },
+  addExistingPrecedingTitle:(precedingTitle) => {
+    cy.do([
+      Button({ id: 'find-instance-trigger' }).click(),
+    ]);
+    InventoryInstanceModal.searchByTitle(precedingTitle);
+    InventoryInstanceModal.selectInstance();
+  },
+  saveAndClose:() => {
+    cy.do(saveAndCloseButton.click());
```
Comment (dmitriyfolio) [inline cypress/support/fragments/inventory/InventoryInstanceEdit.js:108]: fix linter issue

File: cypress/support/fragments/inventory/InventoryInstanceEdit.js [hunk 4]
```diff
@@ -82,7 +85,27 @@ export default {
...
+
+    cy.do([
+      Button('Add preceding title').click(),
+      TextArea({ name:  `${fieldNamePref}.title` }).fillIn(precedingTitle),
+      TextField({ name: `${fieldNamePref}.isbn` }).fillIn(isbn),
+      TextField({ name: `${fieldNamePref}.issn` }).fillIn(issn),
+    ]);
+  },
+  addExistingPrecedingTitle:(precedingTitle) => {
+    cy.do([
```
Comment (dmitriyfolio) [inline cypress/support/fragments/inventory/InventoryInstanceEdit.js:101]: array is not needed in this case

File: cypress/support/fragments/inventory/holdingsMove/inventoryInstanceSelectInstanceModal.js [hunk 5]
```diff
@@ -16,6 +16,15 @@ export default {
     cy.do(rootModal.find(Button('Search')).click());
     cy.expect(rootModal.find(MultiColumnList()).has({ rowCount:1 }));
   },
+  searchByTitle(title) {
+    cy.expect(rootModal.find(HTML(including('Choose a filter or enter a search query to show result'))).exists());
+    cy.do([
+      rootModal.find(SearchField('Search field index')).selectIndex('Title (all)'),
```
Comment (dmitriyfolio) [inline cypress/support/fragments/inventory/holdingsMove/inventoryInstanceSelectInstanceModal.js:22]: move SearchField('Search field index') into common constant

File: cypress/support/fragments/inventory/holdingsMove/inventoryInstanceSelectInstanceModal.js [hunk 6]
```diff
@@ -16,6 +16,15 @@ export default {
...
+      rootModal.find(SearchField('Search field index')).fillIn(title),
+      rootModal.find(Button('Search')).click(),
```
Comment (dmitriyfolio) [inline cypress/support/fragments/inventory/holdingsMove/inventoryInstanceSelectInstanceModal.js:24]: the same

**Comment by mariia-aloshyna** [comment]
Hi @dmitriyfolio and @MaksimTerekhov All comments are resolved, take a look, please

File: cypress/support/api/inventory.js [hunk 7]
```diff
@@ -81,20 +91,6 @@ Cypress.Commands.add('getInstanceTypes', (searchParams) => {
     });
 });
 
-// TODO: move to related fragment
```
Comment (dmitriyfolio) [inline cypress/support/api/inventory.js:84]: what the reason to remove createInstanceType?

**Review by dmitriyfolio — APPROVED** [review]

**Comment by mariia-aloshyna** [comment]
Hi @MaksimTerekhov can this be merged?

**Comment by MaksimTerekhov** [comment]
Hi @mariia-aloshyna 
We are working with 1 approve temporarily now. You can  merge your branch when you get at least one approve :)

**Review by MaksimTerekhov — APPROVED** [review]

**Pull request merged**